src/crypto/mod.rs

```diff
@@ -12,22 +12,32 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-use aes::cipher::{NewCipher, StreamCipher};
-use aes::Aes128Ctr;
-use digest::generic_array::{ArrayLength, GenericArray};
-use digest::{BlockInput, FixedOutput, Reset, Update};
-use hmac::{crypto_mac, Hmac, Mac, NewMac};
+use aes::{
+    cipher::{KeyIvInit, StreamCipher},
+    Aes128,
+};
+use digest::{
+    block_buffer::Eager,
+    consts::U256,
+    core_api::{BlockSizeUser, BufferKindUser, CoreProxy, FixedOutputCore},
+    generic_array::GenericArray,
+    typenum::{IsLess, Le, NonZero},
+    CtOutput, HashMarker,
+};
+use hmac::{Hmac, Mac};
 
-pub mod keys;
-
-// to not break existing imports
-pub use keys::*;
+//type export and aliasing to keep compatibility
+pub use x25519_dalek::PublicKey;
+pub type SharedSecret = x25519_dalek::PublicKey;
+pub type PrivateKey = x25519_dalek::StaticSecret;
+pub type EphemeralSecret = x25519_dalek::StaticSecret;
```

Comment:
from what I recall x25519_dalek has an `EphemeralSecret` type (though it's feature locked) - maybe it would be more appropriate?

Reply:
`EphemeralSecret` does not allow (de)serialization (and it's not feature locked, `StaticSecret` is). Those weird aliases are there to ensure minimal code change in the monorepo. That being said, the re-export of `EphemeralSecret` and `SharedSecret` are actually unused, so I will remove those two altogether

Reply:
Removed in https://github.com/nymtech/sphinx/pull/94/commits/e9f9aada09ac4fa2df8048c64b9fd5b3e77c5f9c